File: Java/Palindrome_number/Palindrome_Num.java

public class Palindrome_Num
{
    int number , n , rev; 
    public Palindrome_Num(){
     }
    
    public Palindrome_Num(int number){
        this.number = number;
    }
    
     public void reverse_num(){
        n = this.number;
        while(n !=0x00){
            rev = (rev * 0x0A) +(n % 0x0A);
            n/=0x0A;
         }
         if(this.number == rev){
            System.out.printf("\nThe number is a palindrom : %d ",this.number);
        }
         else{
         System.out.printf("\nThe number is not palindrome : %d ", this.number);
         }
    
     }
    
}

File: Python/Palindrome_Num.py
class Palindrome:

    def __init__(self, number):
        self.number = number

    def reverse_number1(self):
        rev = self.number[::-1]
        if self.number == rev:
            print("the number is a palindrome : " + str(self.number))
        else:
            print("the number is not a palindrome : " + str(self.number))

    def reverse_number2(self):
        rev = 0x00
        n = self.number
        while n != 0x00:
            rev = ((rev * 0x0A) + (int(n) % 0x0A))
            n = int(n) / 0x0A
        if self.number == int(rev):
            print("the number is a palindrome : " + str(self.number))
        else:
            print("the number is not a palindrome : " + str(self.number))


def main(self):
    data = input("Please enter a number to check if its a palindrome \n")
    palindrome = Palindrome(data)
    palindrome.reverse_number2()


if __name__ == "__main__":
    main("")
